Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1170 @@
+            return ConditionResult(should_skip=False)
+
+
+class OnlyOneRNPerPackCondition(MetadataCondition):
+    def generate_skip_reason(self, rn_files: list, **kwargs) -> str:    # type: ignore[override]
+        """
+        Args:
+            rn_files: release notes files for the pack in current pr.
+        Returns: Reason why the condition failed, and pr skipped.
+        """
+        return SkipReason.MORE_THAN_ONE_RN.format(self.pack, [str(f) for f in rn_files])
+
+    def _check(
+        self, previous_result: Optional[ConditionResult] = None, **kwargs
+    ) -> ConditionResult:
+        """Checks that only one release notes files per pack was added.
+        Args:
+            previous_result: previous check artifacts.
+
+        Returns(ConditionResult): whether the condition check pass,
+            or we should skip this pr from auto-bumping its release notes, with the reason why to skip.
+        """
+        pack_new_rn_files = [
+            Path(f.filename)
+            for f in self.pr.get_files()
+            if f.status == "added"
+            and RELEASE_NOTES_DIR in Path(f.filename).parts
+            and self.pack in Path(f.filename).parts
+            and "md" in Path(f.filename).suffix
+        ]
+        if len(pack_new_rn_files) != 1:
+            return ConditionResult(
+                should_skip=True, reason=self.generate_skip_reason(pack_new_rn_files)
+            )
+        else:
+            return ConditionResult(
+                should_skip=False, pack_new_rn_file=pack_new_rn_files[0]
+            )
+
+
+class SameRNMetadataVersionCondition(MetadataCondition):
+    def __init__(
+        self, pack: str, pr: PullRequest, git_repo: Repo, branch_metadata: Dict
+    ):
+        super().__init__(
+            pack=pack, pr=pr, git_repo=git_repo, branch_metadata=branch_metadata
+        )
+
+    def generate_skip_reason(       # type: ignore[override]
+        self, rn_version: Version, metadata_version: Version, **kwargs
+    ) -> str:
+        """
+        Args:
+            rn_version: version of the release notes.
+            metadata_version: version of the pack in the metadata file.
+        Returns: Reason why the condition failed, and pr skipped.
+        """
+        return SkipReason.DIFFERENT_RN_METADATA_VERSIONS.format(
+            self.pack, rn_version, metadata_version
+        )
+
+    def _check(
+        self, previous_result: Optional[ConditionResult] = None, **kwargs
+    ) -> ConditionResult:
+        """Checks if the new Release Notes has the same version as pack metadata version.
+        Args:
+            previous_result: previous check artifacts.
+
+        Returns(ConditionResult): whether the condition check pass,
+            or we should skip this pr from auto-bumping its release notes, with the reason why to skip.
+        """
+        branch_pack_metadata_version = Version(
+            self.branch_metadata.get(Metadata.CURRENT_VERSION, self.DEFAULT_VERSION)
+        )
```

Comment:
this is repeated multiple times, worth a method

Reply:
It is one line.. the method will be one line also.

Reply:
it's repeated 3 times, if there's a bug, we'll have to fix it 3 times

Reply:
It is only "get" from the dictionary, why it can have a bug?

Reply:
If it ever changes, it'll have to be changed thrice.